### ISSUE TITLE
STM32L1 stuck in WFI after flash error

### ISSUE THREAD

**Issue by tomasvr** [blackmagic-debug/blackmagic#955]
Dear community,
I have been using a BMP to program and debug the stm32l162 processor with GDB. Most of the time this works without problems, but now I have ran into a problem during flashing which has left the mcu unresponsive. 

I was trying to flash new firmware and during the load phase the following error occured:

`Error writing data to flash`

This has happened before, but usually I would just try to flash again and that would work.
This time however, when I retry to flash firmware onto the mcu I keep getting the following response:

```
Target voltage: 1.79V
Timeout during scan. Is target stuck in WFI?
Timeout during scan. Is target stuck in WFI?
SW-DP scan failed!
gdb/gdb.gdb:3: Error in sourced command file:
.gdb/target:4: Error in sourced command file:
Attaching to Remote target failed
```

It seems to me like the MCU is stuck in a faulty state in which it is unresponsive to the SWD lines, possibly because of the error during writing to flash in the load phase. 

**what I've tried so far:**

- Replugging the power of the device
- Replugging all the BMP cables
- Resetting the device through `hard_srst`
- enabling `connect_srst`

None of these seem to help. Since I currently don't have access to another set of cables/probe/target I cannot verify whether this problem is exclusive to this single board.  

Any help is welcome. Thank you in advance!

**Comment by UweBonnes** [comment]
Please compile hosted. First run "blackmagic -t", If it fails, run, post the output. If it syucceeds, try to flash "blackmagic <bla.bin>.  Also state if WFI/WFE or deep sleep is involved, or a remap of the SWJ pins.

**Comment by tomasvr** [comment]
Sorry, I accidentally pressed enter while my post was still incomplete and you had already responded. 

I do make use of deep sleep and WFI. My first thought was also that the MCU might immediately enter WFI after boot which means it would be unreachable for the probe. Thus I tried to enable the `connect_srst` option but that had no effect. Furthermore I do not see any output on my serial debug line so I suspect the application is not being run at all. 
Lastly, I am not aware of any SWJ pin remapping.

I will try your suggestion, thank you!

**Comment by UweBonnes** [comment]
Just a remainder: connect_srst needs NRST connected. What BMP platforms do you use?

**Comment by dragonmux** [comment]
👋🏼 Apologies that this one appears to have gotten away from us. We believe that our issue may have been fixed in some of the work we did in v1.10 or since on `main` during the overhaul of the STM32L0/L1 support. Please give the latest firmware a test and let us know how you get on.

**Comment by dragonmux** [comment]
Given it's been a month and we've still not heard anything, we're going to close this as fixed. If it turns out it is not fixed, please let us know and we'll re-open the issue.
